axelrod/tests/strategies/test_player.py

```diff
@@ -646,28 +627,53 @@ def versus_test(
         player2,
         expected_actions1,
         expected_actions2,
+        turns=None,
         noise=None,
         seed=None,
+        match_attributes=None,
+        attrs=None
     ):
         """Tests a sequence of outcomes for two given players."""
         if len(expected_actions1) != len(expected_actions2):
-            raise ValueError("Mismatched History lengths.")
-        if seed:
-            axl.seed(seed)
-        turns = len(expected_actions1)
-        match = axl.Match((player1, player2), turns=turns, noise=noise)
+            raise ValueError("Mismatched Expected History in TestMatch.")
+        if not turns:
+            turns = len(expected_actions1)
+
+        match = axl.Match(
+            (player1, player2), turns=turns, noise=noise, seed=seed,
+            match_attributes=match_attributes)
         match.play()
-        # Test expected sequence of play.
-        for i, (outcome1, outcome2) in enumerate(
-            zip(expected_actions1, expected_actions2)
-        ):
-            player1.play(player2)
-            self.assertEqual(player1.history[i], outcome1)
-            self.assertEqual(player2.history[i], outcome2)
+
+        # Test expected sequence of plays from the match is as expected.
+        for i, (play, expected_play) in enumerate(zip(player1.history, expected_actions1)):
+            self.assertEqual((i, play), (i, expected_play))
+        for i, (play, expected_play) in enumerate(zip(player2.history, expected_actions2)):
+            self.assertEqual((i, play), (i, expected_play))
+
+        # Test final player attributes are as expected
+        if attrs:
+            for attr, value in attrs.items():
+                self.assertEqual(getattr(player1, attr), value)
+
+    def search_seeds(self, *args, **kwargs):  # pragma: no cover
+        """Search for a seed that will pass the test. To use to find a new seed
+        for a versus_test, change self.versus_test to self.search_seeds
+        within a TestPlayer or TestMatch class.
```

Comment:
Nice addition. Could we document this a bit more, one of the two options perhaps:

1. Add an example to the docstring.
2. Specifically add a discussion of this to the contributions docs

Certainly doesn't have to be done in this PR: we can open an issue and do that later :)

Reply:
I was thinking that we'd update the library to Python 3.7 + 3.8 or start using `attrs`. Then I can add back in the [code](https://github.com/Axelrod-Python/Axelrod/pull/1288/commits/5128e4d8f6762986e5449564349d8874f7a54802) I used to watch `versus_test`, record all the matches, and automatically search for new seeds using this function, which could all be documented in the contributions section.

Reply:
Sounds great!

Reply:
Opened  #1361